### DISCUSSION TITLE
Using Button to Create Note From Template with Dynamic / Calculated File Name

### DISCUSSION THREAD

**Discussion by cacharbe** [mProjectsCode/obsidian-meta-bind-plugin#228] [Q&A]
Per the suggestion on his "Buttons" page, I've followed the advice of @shabegom and am trying to start using meta bind.

I currently use the buttons plug in to create a file based on a file template and dynamically generate the filename on button click using templater  (<% tp.user.GetSomeDataCalculatedByInfoInVault(tp) %> ) in the button code.

Is this pattern even possible using Meta Bind Button?  If someone can point me in the right direction or if you have an example, I'd be greatly appreciative.

**Comment by I-Pch** [comment]
I just very recently tested the use of Meta Bind Buttons with Templater and there are ways to get there, @cacharbe  😊 ...

If you use a Meta Bind Button of `type` [templaterCreateNote](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/buttons/#templater-create-note-action), in the Templater's template applied on the creation of the new note, you could add a [tp.file.rename()](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilerenamenew_title-string) or [tp.file.move()](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilemovenew_path-string-file_to_move-tfile) somewhere to simply rename or move and rename the brand new note with your desired title ... 

So, instead of trying to use your user script directly in the button `fileName` part, you would give an actual name to the note from the template applied on the file's creation 😊 . 

Another way would be to instead of using a button of type `templaterCreateNote` to use a button of type [command](https://www.moritzjung.dev/obsidian-meta-bind-plugin-docs/guides/buttons/#command-action)...
 
For this to work, you'd need to add a hotkey to the template you wish to use in the button in Templater's settings (you actually don't need to effectively add a hotkey to the template, just say in Templater's setting that "yes, the template could be triggered through a hotkey") which makes the template available as a `command` in Obsidian's command palette ... 
Then, by using a `Command` type of button, the button would call that command and trigger the desired template 😊 ...

The thing with this method though 😅, is that you'd need to adapt the "triggered" template so the new note would be created through the template (using [tp.file.create_new()](https://silentvoid13.github.io/Templater/internal-functions/internal-modules/file-module.html#tpfilecreate_newtemplate-tfile--string-filename-string-open_new-boolean--false-folder-tfolder) for example) instead of the button 😊.

Depending on your needs/workflow one or the other could potentially work for your use case 😇 .

**Reply by cacharbe** [reply]
Thank you for the guidance, I appreciate the thoughtfulness.  I will start digging into these patterns.

**Reply by I-Pch** [reply]
My pleasure @cacharbe 😊 !

**Reply by mgrecar** [reply]
Oh man, this is _exactly_ what I was looking for, thanks so much!

**Reply by I-Pch** [reply]
My pleasure 😊 !
